gradle/new-code-coverage.gradle.kts

```
@@ -0,0 +1,227 @@
+    // referenced: a script applied with apply(from = ...) has no type-safe task accessors.
+    dependsOn("test")
+
+    val report = layout.buildDirectory.file("reports/jacoco/test/jacocoTestReport.xml")
+    val sourceRoot = "src/main/java/"
+    // The prefix carries a trailing slash so that removePrefix cuts cleanly; in a sentence the
+    // path reads better without it.
+    val sourceRootName = sourceRoot.trimEnd('/')
+    val baseRef = (findProperty("newCodeCoverage.baseRef") ?: "origin/main").toString()
+    // A share, not a percentage. Written as 80 it would ask for 8000% and fail every branch,
+    // with a figure in the comment that says nothing about what went wrong.
+    val minimum = (findProperty("newCodeCoverage.minimum") as String?)?.let { given ->
```

detekt (1.23.8)

[warning] Use the safe cast ('as? String') instead of 'as String?'.

(detekt.potential-bugs.CastToNullableType)